browse/controllers/files/dissemination.py

```diff
@@ -83,27 +85,38 @@ def src_resp_fn(format: FileFormat,
                         etag=file.etag,
                         last_modified=file.updated,
                         size=file.size).make_response()
+
     suffixes = Path(file.name).suffixes
     if not arxiv_id.is_old_id:
         suffixes.pop(0)  # get rid of .12345
     filename = download_file_base(arxiv_id, version) + "".join(suffixes)
+    resp.headers["Content-Disposition"] = f"attachment; filename=\"{filename}\""
 
     add_mimetype(resp, file.name)
-    resp.headers["Content-Disposition"] = f"attachment; filename=\"{filename}\""
     add_time_headers(resp, file, arxiv_id)
     return resp  # type: ignore
 
+def pdf_resp_fn(format: FileFormat,
+                 file: FileObj,
+                    arxiv_id: Identifier,
+                    docmeta: DocMetadata,
+                    version: VersionEntry,
+                    extra: Optional[str] = None) -> Response:
+    """funciton to make a `Response` for a PDF."""
+    resp = default_resp_fn(format, file, arxiv_id, docmeta, version, extra)
+    filename = f"{arxiv_id.filename}v{version.version} .pdf"
```

Comment:
Extra space befor .pdf is probably not what you want.

Reply:
Thanks!